test/e2e/stackdriver_monitoring.go

```diff
@@ -39,5 +46,148 @@ var _ = framework.KubeDescribe("Stackdriver metrics", func() {
 	})
 })
 
-func testStackdriverMonitoring() {
+var (
+	stackdriverMetrics = []string{
+		"uptime",
+		"cpu/reserved_cores",
+		"cpu/usage_time",
+		"cpu/utilization",
+		"memory/bytes_total",
+		"memory/bytes_used",
+		"memory/page_fault_count",
+		"disk/bytes_used",
+		"disk/bytes_total",
+	}
+)
+
+func testStackdriverMonitoring(c clientset.Interface) {
+	_, err := expectedPods(c)
+	time.Sleep(time.Minute * 5)
+	projectId := framework.TestContext.CloudConfig.ProjectID
+	for _, metric := range stackdriverMetrics {
+		_, err = fetchTimeSeries(projectId, metric)
+		if err != nil {
+			framework.Failf("Error fetching %v, %v", metric, err)
+		}
+	}
+}
+
+func createMetricFilter(metric string) string {
+	return fmt.Sprintf("metric.type=\"container.googleapis.com/container/%s\"", metric)
+}
+
+func fetchTimeSeries(projectId string, metric string) (*gcm.ListTimeSeriesResponse, error) {
+	ts := google.ComputeTokenSource("")
+
+	// Hack for running tests locally
+	// If this is your use case, create application default credentials:
+	//
+	// $ gcloud auth application-default login
+	//
+	// and uncomment following lines:
+	//
+	// ts, err := google.DefaultTokenSource(oauth2.NoContext)
+	// framework.Logf("Couldn't get application default credentials, %v", err)
+	// if err != nil {
+	// 	framework.Failf("Error accessing application default credentials, %v", err)
+	// }
+
+	client := oauth2.NewClient(oauth2.NoContext, ts)
+	gcmService, err := gcm.New(client)
+	if err != nil {
+		return nil, err
+	}
+	now := time.Now()
+	start := now.Add(time.Duration(-300) * time.Second)
+	response, err := gcmService.Projects.TimeSeries.
+		List(fullProjectName(projectId)).
+		Filter(createMetricFilter(metric)).
+		IntervalStartTime(start.Format(time.RFC3339)).
+		IntervalEndTime(now.Format(time.RFC3339)).
+		Do()
+	if err != nil {
+		return nil, err
+	}
+	if len(response.TimeSeries) > 0 {
+		return response, nil
+	}
+	return nil, fmt.Errorf("No uptime timeseries found")
+}
+
+func fullProjectName(name string) string {
+	return fmt.Sprintf("projects/%s", name)
+}
+
+func expectedPods(c clientset.Interface) ([]string, error) {
```

Comment:
This is a copy/paste from `monitoring.go`. Why copy/paste? Why do you need this?